### ISSUE TITLE
Label encoding categorical variables

### ISSUE THREAD

**Issue by vopani** [mlr-org/mlr#1263]
Is there a functionality to label encode categorical variables? Like how Scikit has.

I know its probably easy, using factors, but was wondering if this exists or is planned to be introduced (something similar to 'createDummyFeatures'), since I was unable to locate it?

For reference:
**Scikit** - http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html
**Stackoverflow** - http://stackoverflow.com/questions/38620424/label-encoder-functionality-in-r

**Comment by jakob-r** [comment]
At the moment this is not directly implemented in *mlr* and it's also not directly planned. 

If a learner supports factors but only by having them as numerical integers we take care of that internally for some learners. However this might change in the future. But we want to be careful to not have this future misused. 

Until then you can [write yourself a pretty basic preprocessing wrapper](https://mlr-org.github.io/mlr-tutorial/devel/html/preproc/index.html#writing-a-custom-preprocessing-wrapper) which takes care of exactly that. Of course we are happy if you share your results :wink: 

Out of curiosity: For which learner do you want to use this feature?

**Comment by rohanrao91** [comment]
This isn't learner specific. Just a pre-processing step thats popularly used to handle features with large number of categories, before plugging it into a model/learner.

I'll check the custom preprocessing wrapper.

To answer your question: Say I am using random-forest. It doesn't handle large number of factors (like 100+). So, I'd want to transform it before plugging into RF. Dummy features don't always scale well.

**Comment by jakob-r** [comment]
I know that this isn't learner specific. But by converting it to an integer you will now in the majority of cases let the learner think that there is an order in your previously unordered categorical feature. 
There might be clever ways to apply the integer representations to guide the random forest at least a bit. But it can also render the feature useless.

**Comment by coforfe** [comment]
Hello,

My understanding is that you are using `randomForest` package, if so. You can use the alternative implementation of `ranger` which is much faster and it has a new parameter _respect.unordered.factor_ which can handle higher number of levels like your case.

For a complete discussion of how to use it you can read this blog entry:
http://www.win-vector.com/blog/2016/05/on-ranger-respect-unordered-factors/#comment-66275

**Comment by zmjones** [comment]
Are they just doing random evaluation of some of the levels in the factor? I don't see how else you could speed this up.
